trin-execution/src/engine/rpc.rs

```diff
@@ -0,0 +1,146 @@
+    ) -> RpcResult<TransitionConfiguration>;
+
+    #[method(name = "forkchoiceUpdatedV1")]
+    async fn fork_choice_updated_v1(
+        &self,
+        fork_choice_state: ForkchoiceState,
+        payload_attributes: Option<PayloadAttributes>,
+    ) -> RpcResult<ForkchoiceUpdated>;
+
+    #[method(name = "forkchoiceUpdatedV2")]
+    async fn fork_choice_updated_v2(
+        &self,
+        fork_choice_state: ForkchoiceState,
+        payload_attributes: Option<PayloadAttributes>,
+    ) -> RpcResult<ForkchoiceUpdated>;
+
+    #[method(name = "forkchoiceUpdatedV3")]
+    async fn fork_choice_updated_v3(
+        &self,
+        fork_choice_state: ForkchoiceState,
+        payload_attributes: Option<PayloadAttributes>,
+    ) -> RpcResult<ForkchoiceUpdated>;
+
+    #[method(name = "getPayloadBodiesByHashV1")]
+    async fn get_payload_bodies_by_hash_v1(
+        &self,
+        block_hashes: Vec<B256>,
+    ) -> RpcResult<ExecutionPayloadBodiesV1>;
+
+    #[method(name = "getPayloadBodiesByHashV2")]
+    async fn get_payload_bodies_by_hash_v2(
+        &self,
+        block_hashes: Vec<B256>,
+    ) -> RpcResult<ExecutionPayloadBodiesV2>;
+
+    #[method(name = "getPayloadBodiesByRangeV1")]
+    async fn get_payload_bodies_by_range_v1(
+        &self,
+        start: u64,
+        count: u64,
+    ) -> RpcResult<ExecutionPayloadBodiesV1>;
+
+    #[method(name = "getPayloadBodiesByRangeV2")]
+    async fn get_payload_bodies_by_range_v2(
+        &self,
+        start: u64,
+        count: u64,
+    ) -> RpcResult<ExecutionPayloadBodiesV2>;
+
+    #[method(name = "getPayloadV1")]
+    async fn get_payload_v1(&self, payload_id: PayloadId) -> RpcResult<ExecutionPayloadV1>;
+
+    #[method(name = "getPayloadV2")]
+    async fn get_payload_v2(&self, payload_id: PayloadId) -> RpcResult<ExecutionPayloadV2>;
+
+    #[method(name = "getPayloadV3")]
+    async fn get_payload_v3(&self, payload_id: PayloadId) -> RpcResult<ExecutionPayloadV3>;
+
+    #[method(name = "getPayloadV4")]
+    async fn get_payload_v4(&self, payload_id: PayloadId) -> RpcResult<ExecutionPayloadV4>;
+
+    #[method(name = "newPayloadV1")]
+    async fn new_payload_v1(&self, payload: ExecutionPayloadV1) -> RpcResult<PayloadStatus>;
+
+    #[method(name = "newPayloadV2")]
+    async fn new_payload_v2(&self, payload: ExecutionPayloadInputV2) -> RpcResult<PayloadStatus>;
+
+    #[method(name = "newPayloadV3")]
+    async fn new_payload_v3(
+        &self,
+        payload: ExecutionPayloadV3,
+        versioned_hashes: Vec<B256>,
+        parent_beacon_block_root: B256,
+    ) -> RpcResult<PayloadStatus>;
+
+    #[method(name = "newPayloadV4")]
+    async fn new_payload_v4(
+        &self,
+        payload: ExecutionPayloadV4,
+        versioned_hashes: Vec<B256>,
+        parent_beacon_block_root: B256,
+    ) -> RpcResult<PayloadStatus>;
+}
+
+/// A subset of Eth JSON-RPC endpoints under the Engine Api's JWT authentication
+#[rpc(client, server, namespace = "eth")]
```

Comment:
nit: While it's only a subset, does it make sense to move this into: https://github.com/ethereum/trin/blob/master/ethportal-api/src/eth.rs ?

Reply:
> nit: While it's only a subset, does it make sense to move this into: https://github.com/ethereum/trin/blob/master/ethportal-api/src/eth.rs ?

I think both of the traits I added should be next to each other, because they will only be used together.

I think we should have both `EngineEthApi` and `EthApi` trait, because they are used in 2 different contexts